posthog/tasks/calculate_cohort.py

```diff
@@ -13,12 +13,15 @@
 
 logger = logging.getLogger(__name__)
 
+MAX_AGE_MINUTES = 15
 
-def calculate_cohorts(max_age_minutes: int = 15) -> None:
-    start_time = time.time()
+
+def calculate_cohorts() -> None:
+    # This task will be run every minute
+    # Every minute, grab a few cohorts off the list and execute them
     for cohort in Cohort.objects.filter(
-        Q(is_calculating=False) | Q(last_calculation__lte=timezone.now() - relativedelta(minutes=max_age_minutes))
-    ).order_by("id"):
+        Q(is_calculating=False) | Q(last_calculation__lte=timezone.now() - relativedelta(minutes=MAX_AGE_MINUTES))
+    ).order_by("last_calculation")[0:15]:
```

Comment:
Does .order_by put nulls first or last? If last some cohorts may never be processed.

Reply:
Good question. By default I think yes, but I've made it super explicit now